imagebuildah/build.go

```diff
@@ -1154,7 +1165,7 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) error
 	// the removal of intermediate/build containers will be handled by the
 	// defer statement above.
 	if b.removeIntermediateCtrs && (b.layers || b.noCache) {
-		return stageExecutor.Delete()
+		return b.deleteSuccessfulIntermediateCtrs()
```

Comment:
We might want to eat the error here and report a different error.
Something like
```
if err = b.deleteSuccessfulIntermediateCtrs(); err != nil {
           return errors.Error("Failed to cleanup intermediate Containers");
}
```

Otherwize the `LastError` will be reported twice.

Reply:
Ooh, right. fixed. Thanks!